[Bug 1268572] The big download button overhaul

alexgibson opened the pull request:
## Description
- Update button style of general CTA buttons ([Bug 1262099](https://bugzilla.mozilla.org/show_bug.cgi?id=1262099))
- Decouple “other” links from download button & update link copy ([Bug 1259211](https://bugzilla.mozilla.org/show_bug.cgi?id=1262099))
- Allow custom download button label ([Bug 1089725](https://bugzilla.mozilla.org/show_bug.cgi?id=1089725))
- Update download pages to show build links even if we're unable to verify that the user meets system requirements ([Bug 1116166](https://bugzilla.mozilla.org/show_bug.cgi?id=1116166))
-  Change download button style and logic ([Bug 1258662](https://bugzilla.mozilla.org/show_bug.cgi?id=1258662))
-  ~~Remove outdated styleguide pages ([Bug 1268847](https://bugzilla.mozilla.org/show_bug.cgi?id=1268847))~~

## Bugzilla
Tracking bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1268572

## Testing
https://www-demo4.allizom.org/en-US/

## Checklist
- [x] Requires l10n changes.
- [x] Related functional & integration tests passing.

alexgibson commented on bedrock/firefox/helpers.py:
```diff
@@ -56,32 +56,61 @@ def ios_builds(channel, builds=None):
-                     platform='all', dom_id=None, locale=None, simple=False,
-                     force_direct=False, force_full_installer=False,
-                     force_funnelcake=False, check_old_fx=False):
+def firefox_footer_links(ctx, channel='release', platform='all'):
```
Note for self: we should probably have tests for this?

alexgibson replied:
Added tests in a follow up commit 👍

alexgibson commented on bedrock/firefox/templates/firefox/choose.html:
```diff
@@ -0,0 +1,68 @@
+{# This Source Code Form is subject to the terms of the Mozilla Public
```
Note to self: this page has previously been removed from bedrock, so looks like this may have been added back accidentally during a rebase. Need to check to make sure any other files haven't been reintroduced.

alexgibson commented on bedrock/firefox/templates/firefox/developer.html:
```diff
@@ -217,16 +217,22 @@
+      {{ email_newsletter_form('app-dev', 'Firefox Apps & Hacks', button_class='light') }}
     {% elif LANG.startswith('es-') %}
-      {{ email_newsletter_form('app-dev', 'Firefox Apps y Hacks', button_class='white') }}
+      {{ email_newsletter_form('app-dev', 'Firefox Apps y Hacks', button_class='light') }}
```
`Firefox Apps y Hacks`?

alexgibson commented on bedrock/firefox/templates/firefox/releases/release-notes.html:
```diff
@@ -310,10 +310,19 @@
 </footer>
 {% endblock %}
 
+<p>Some info: {{ release }}</p>
```
Looks like this is https://bugzilla.mozilla.org/show_bug.cgi?id=1268318

alexgibson commented on media/css/sandstone/buttons.less:
```diff
@@ -345,267 +343,6 @@ html[lang^='en'] {
     }
 }
 
-.download-button .download-link {
```
Note to self: various parts of `buttons.less` are being removed here, but the file itself no longer seems to be used in sandstone with this change. Do we still need any of it?

alexgibson commented on bedrock/styleguide/urls.py:
```diff
@@ -91,8 +88,6 @@
 
 if settings.DEV:
     all_children.extend((
-        PageNode('All Buttons', path='all-download-buttons',
```
Note to self: this page needs reinstating and should not be removed

schalkneethling commented:
Picked a small item: home-b.html needs the newsletter button color set to 'light'

alexgibson commented on bedrock/firefox/templates/firefox/pocket.html:
```diff
@@ -42,13 +42,13 @@
 
       {# Old Firefox or Not Firefox #}
-      <p class="main-cta" id="cta-download"><a href="{{ url('firefox.new') }}?scene=2" class="button-flat-green">{{ _('Download Firefox') }}</a></p>
+      <p class="main-cta" id="cta-download"><a href="{{ url('firefox.new') }}?scene=2" class="button green">{{ _('Download Firefox') }}</a></p>
```
Note to self: any reason we can't make this a real download button, now we can pass in custom strings?

alexgibson commented on bedrock/firefox/templates/firefox/sync.html:
```diff
@@ -132,19 +132,19 @@
 
         <div class="show-fx-28-older">
-          <a href="{{ url('firefox.new') }}?scene=2" class="sync-button green" data-interaction="button click" data-element-action="Sync CTA" id="cta-firefox">
+          <a href="{{ url('firefox.new') }}?scene=2" class="button green" data-interaction="button click" data-element-action="Sync CTA" id="cta-firefox">
```
Same here, can we use a real download button?

alexgibson commented on bedrock/mozorg/templates/mozorg/home/home.html:
```diff
@@ -156,7 +156,7 @@ <h2 class="go">{{ _('Download Firefox') }}</h2>
           </div>
           <div class="secondary">
-            {{ download_firefox(simple=true, dom_id='promo-download-fx-button') }}
+            {{ download_firefox(alt_copy=_('Free Download'),button_color='transparent-light',dom_id='promo-download-fx-button') }}
```
Nit: space here

alexgibson commented:
OK, I'm marking this as ready for review. @jpetto r? (when you have enough time for this).

I'm going to keep the do-not-merge label for now, as I'd like to push this to demo after review and run the integration tests one last time before we merge.

alexgibson commented on bedrock/mozorg/templates/mozorg/plugincheck.html:
```diff
@@ -97,7 +97,7 @@ <h1 class="large">{{_('Check Your Plugins')}}</h1>
         <h2>{{_('Looks like you’re using an older version of Firefox.')}}</h2>
         <div class="desktop download-button-wrapper">
-          {{ download_firefox(simple=true, check_old_fx=true) }}
+          {{ download_firefox(check_old_fx=true) }}
```
Note to self: seems `check_old_fx` may be redundant now? The corresponding code no longer seems to do anything, since buttons no longer have subtitles: https://github.com/mozilla/bedrock/blob/master/media/js/base/global.js#L52

This seems to be the only page where this is used. Should we just remove the code?

alexgibson replied:
Double checked with @schalkneethling on IRC and agrees we should probably remove this. Going to make it so.

alexgibson commented on bedrock/firefox/templates/firefox/channel.html:
```diff
@@ -116,6 +116,11 @@ <h2 class="channel-title channel-title-aurora">{{ high_res_img('firefox/channel/
       </p>
+      {% if l10n_has_tag('channel_aurora_android') %}
+        <p class="info">
+          <a rel="external" href="https://support.mozilla.org/kb/will-firefox-work-my-mobile-device">{{_('Will Firefox work on my mobile device?')}}</a>
```
@flodolo - note I had to add one new string to this PR since it was originally opened. Did so behind a tag here.

flodolo replied:
Thanks, noted

alexgibson commented:
Putting this back to WIP for now, found there are still quite a few occurances of `button-flat`, `button-flat-dark` and `button-flat-green` in the branch, for which styles no longer exist.

alexgibson commented:
Ok, I think ducks are in a line - opening this up for review again.

jpetto commented:
Privacy link is really low contrast for non-Fx users on `/firefox/accounts/`.

![screen shot 2016-05-09 at 5 18 22 pm](https://cloud.githubusercontent.com/assets/317532/15129875/3bc74eb0-160a-11e6-8715-b6970b256cb4.png)

jpetto commented:
Similar contrast issue on `/firefox/android/faq/`:

![screen shot 2016-05-09 at 5 21 14 pm](https://cloud.githubusercontent.com/assets/317532/15129917/81ce3572-160a-11e6-9d68-cb2a9b9c40f6.png)

jpetto commented:
Rather minor, but for non-Fx users on `/firefox/desktop/`, something about forced width is causing the sticky nav button to lack curved corners and left/right borders:

![screen shot 2016-05-09 at 5 33 44 pm](https://cloud.githubusercontent.com/assets/317532/15130235/6b944fa6-160c-11e6-95bd-8d2126e7dab3.png)

(I would not call this a blocker.)

jpetto commented on bedrock/firefox/templates/firefox/includes/download-button.html:
```diff
@@ -41,73 +41,41 @@
+                 <strong class="download-title">{{ _('Download Firefox') }}</strong>
+               {% endif %}
+             {% endif %}
+          {% endif %}
```
Unless white space would cause an issue, would be nice to not repeat the `<strong class="download-title">` tag for each variation. Maybe try wrapping the logic in the `<strong>` tag instead?

alexgibson replied:
Good suggestion 👍

jpetto commented:
Another non-blocker, but the button layout on `/firefox/organizations/` is pretty bulky:

![screen shot 2016-05-09 at 5 53 10 pm](https://cloud.githubusercontent.com/assets/317532/15130562/fe31b9be-160e-11e6-9b21-8a9de371885a.png)

jpetto commented:
Another small, non-blocker UI nit - button on `/firefox/sync/` for Fx 29 or 30 users isn't centered:

![screen shot 2016-05-09 at 6 02 04 pm](https://cloud.githubusercontent.com/assets/317532/15130719/41b4420a-1610-11e6-88f4-b459c6b0d073.png)

alexgibson commented:
Thanks @jpetto - I've rebased against the latest master now that #4101 merged. Also added a followup commit that should address review comments so far.

jpetto commented on bedrock/styleguide/urls.py:
```diff
@@ -51,9 +51,12 @@
+            PageNode('Forms', path='forms', template='styleguide/websites/sandstone-forms.html'),
             PageNode('Grids', path='grids', template='styleguide/websites/sandstone-grids.html'),
             PageNode('Tables & Lists', path='tables', template='styleguide/websites/sandstone-tables.html'),
+            PageNode('Tabzilla', path='tabzilla', template='styleguide/websites/sandstone-tabzilla.html'),
```
Looks like these URLs (and templates) were removed in #4104. Maybe an accidental regression here?

alexgibson replied:
Ah good catch, the peril of rebasing a huge branch :)

jpetto commented on media/css/firefox/new/common.less:
```diff
@@ -40,6 +40,23 @@
+    display: none;
+}
+
+/* show links based on thge current os */
```
Thge is my favorite character on Game of Thrones. :trollface:

jpetto commented on media/css/firefox/new/redesign/common.less:
```diff
@@ -188,3 +181,20 @@ main {
+.android .fx-footer-links.os_android,
+.ios .fx-footer-links.os_ios {
+    display: block;
+}
```
Should this display logic be made into an include or mixin? You're much more familiar with this entire button system than I, so definitely your call. Just thought I'd ask. :smile:

alexgibson replied:
Hmm, this is only used on two pages (so far), but given the the big splurge of selectors involved I am definitely in favor of reducing error via copy/pasta. A mixin is a fine suggestion 👍

jpetto commented on media/css/firefox/os/firefox-os-2-5.less:
```diff
@@ -286,58 +286,64 @@ section {
+//         text-decoration: none;
+//     }
+// }
+
```
Get rid of all these commented out lines?

jpetto commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -379,6 +379,23 @@ ul.section-items {
+.android .fx-footer-links.os_android,
+.ios .fx-footer-links.os_ios {
+    display: inline-block;
+}
```
Oooh, one more instance of the platform display logic?

jpetto commented on media/css/firefox/sync.less:
```diff
@@ -357,9 +323,11 @@
-    width: 250px;
-    margin: 0 auto;
+.js .show-not-fx .download-button {
+    ul {
```
Any reason to change the nesting here?

jpetto commented on media/css/sandstone/buttons.less:
```diff
@@ -2,612 +2,299 @@
+    background: none;
+    padding: (@baseLine / 2) 36px;
+    border-radius: 4px;
+    .font-size(1.25rem);
```
Use `px` value here instead?

jpetto commented:
Low priority, but seeing a disc on the `<li>` in IE 7:

![screen shot 2016-05-10 at 10 27 32 am](https://cloud.githubusercontent.com/assets/317532/15152175/e451e6b4-1699-11e6-8dea-385f47270276.png)

Seems to show up in all instances, not just on `/firefox/new/`.

jpetto commented:
Tested home and `/firefox/new/` in old IEs and w/out JS in A-grade browsers and all is looking good. Spot checked most files changed in Fx & Chrome and did not see any issues.

A couple small suggestions/fixes/cleanups, but overall this enormous PR is really close! :muscle:

alexgibson commented:
Thanks @jpetto - I've added a followup commit that should hopefully address the second round of review feedback. During the process I also uncovered a bit of unfinished coding for the footer links on `/releasenotes/` pages. I've included that here, so it could do a second pair of eyes. Thanks again!

jpetto commented:
Nice catch on the `/releasenotes/` pages. Code looks good and functions as expected there.

All other issues look to be taken care of as well. Woohoo!

alexgibson commented:
Thanks @jpetto - after running the functional tests against this branch on demo4, assuming everything passes, I think we might be OK to merge this? (gulp).

alexgibson commented:
@jgmize ran the full suite of cross-browser integration tests against demo4:

https://ci.us-west.moz.works/job/bedrock-cross-browser/2/

With the exception of three redirect failures that are unrelated (due to demo4 being on our legacy infra), all tests passed 👍

alexgibson commented:
@jpetto - squashed and rebased. Is it time? 😬

jpetto commented:
No time like the present!

(We also have a couple non-sprint days coming up in which to watch for/fix bugs.)
